src/helpers/GenioCommon.cpp

```diff
@@ -12,15 +12,27 @@
 #include <string>
 #include "Utils.h"
 #include <Path.h>
-
+#include "Log.h"
 namespace Genio
 {
 
 std::string const
 file_type(const std::string& fullpath)
 {
-	BPath path(fullpath.c_str());
-	std::string filename = path.Leaf();
+	BPath path;
+	//should 'support' also the URI format
```

Comment:
We should use BUrl instead

Reply:
Yes you are right. I'll try use BUrl inside all the LSP related class to avoid any string manipulation